oio/directory/meta0.py

```diff
@@ -201,6 +199,62 @@ def __init__(self, meta0_client, replicas=3,
         if self.digits > 4:
             raise ConfigurationException("meta_digits must be <= 4")
         self.min_dist = min_dist
+        if self.min_dist < 1:
+            raise ConfigurationException("min_dist must be >= 1")
+        if self.min_dist > 4:
+            raise ConfigurationException("min_dist must be <= 4")
+        super(Meta0PrefixMapping, self).__init__(
+            meta0_client.conf, ['meta1'], **kwargs)
+
+    def reset(self):
+        super(Meta0PrefixMapping, self).reset()
+
+        for svc_id, svc in self.services.iteritems():
+            location = self.get_loc(svc)
+            if self.min_dist == 1:
+                site = location
+            else:
+                site = '.'.join(location.split('.')[:1-self.min_dist])
+            svc['site'] = site
+            svc['upper_limit'] = 0
+
+    def _compute_upper_limit(self):
```

Comment:
Please stop thinking that prefixing a method name with an underscore prevents you from documenting it ;-)

Here is a description of what I think this method does:
> Compute the right number bases each service should host, based on the total number of bases, the total number of service locations, and the number of services per location.